### PR TITLE
Add transparent borders on brand component

### LINKED ISSUE #646: Brand - Add transparent border for accessibility
**Describe the bug**
When you view the brand component in high contrast mode on PC or if you change the colour preferences in FF you don't see background colours and therefore can't see when the brand component ends/starts.

**To Reproduce**
- Change the colour preferences in FF - See check point 5 https://bbc-news.github.io/accessibility-news-and-you/accessibility-news-and-testers
- View in high contrast mode in PC - See check point 4 https://bbc-news.github.io/accessibility-news-and-you/accessibility-news-and-testers

**Expected behaviour**
When the brand component is in the header add CSS

```
  border-bottom: 0.0625rem solid transparent;
```
When the brand component is in the footer add CSS

```
  border-top: 0.0625rem solid transparent;
```

We will need different behaviour if there is navigation in the header - No border for the brand as we will have a bottom border from the navigation component - Speak to @DenisBBC 

**Screenshots**
Header - High contrast mode on PC without border bottom:
banner-top

Header - High contrast mode on PC with border bottom:
banner-top-with-line

Footer - Colour preferences changed in FF without border:
FF-bottom

Footer - Colour preferences changed in FF with border top and bottom (note bottom not to be applied here):
FF-bottom-with-border

### PR DESCRIPTION
**Pull request by ghost**
Resolves #646 

**Overall change:** Add transparent borders controlled via props to make the start/end of `Brand` component clear in high contrast mode.

**Code changes:**

- Add optional props `borderBottom` and `borderTop`

---

- [X] I have assigned myself to this PR and the corresponding issues
- [X] Tests added for new features
- [ ] Test engineer approval

### CODE REVIEW

File: packages/components/psammead-brand/README.md [hunk 1]
```diff
@@ -34,11 +34,15 @@ The `url` value is the link that points to the frontpage of the service associat
...
 | url | String | no | N/A | `https://www.bbc.co.uk/news` |
 | serviceLocalisedName | String | no | N/A | `'Yoruba'` |
+| borderTop | bool | no | `false` | `true` |
+| borderBottom | bool | no | `false` | `true` |
 
 ## Usage
 
 The typical use-case of this component is at the top of pages in a [`header` element](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/header). When this is done it is recommend that the component is wrapped in a [`banner` role](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/Banner_role). However, a `header` with a `banner` role should only appear once on a page.
 
+When using `Brand` on at the top pages, you should ensure that `borderBottom` prop is set to true. Also when using brand on the footer you should set `borderTop` to true. This ensures that in high contrast mode the start/end of the `Brand` component is visible.
```
Comment (andrew-nowak) [inline packages/components/psammead-brand/README.md]:
```suggestion
When using `Brand` in the header, you should ensure that `borderBottom` prop is set to true. Similarly, when using brand on the footer you should set `borderTop` to true. This ensures that in high contrast mode the top/bottom of the `Brand` component is visible.
```

File: packages/components/psammead-brand/src/index.jsx [hunk 2]
```diff
@@ -31,6 +31,9 @@ const Banner = styled.div`
       conditionallyRenderHeight(svgHeight, PADDING_AROUND_SVG_ABOVE_600PX)};
     padding: 0 ${GEL_SPACING_DBL};
   }
+  ${({ borderTop }) => borderTop && 'border-top: 1px solid transparent;'}
```
Comment (andrew-nowak) [inline packages/components/psammead-brand/src/index.jsx]:
```suggestion
  ${({ borderTop }) => borderTop && 'border-top: 0.0625rem solid transparent;'}
```

File: packages/components/psammead-brand/CHANGELOG.md [hunk 3]
```diff
@@ -3,6 +3,7 @@
 <!-- prettier-ignore -->
 | Version | Description |
 | ------- | ----------- |
+| 4.0.2   | [PR#647](https://github.com/bbc/psammead/pull/647) Add transparent borders with borderTop and borderBottom props |
```
Comment (andrew-nowak) [inline packages/components/psammead-brand/CHANGELOG.md]: This PR adds a new feature, I believe the version bump should be to 4.1.0

File: packages/components/psammead-brand/src/index.jsx [hunk 4]
```diff
@@ -31,6 +31,9 @@ const Banner = styled.div`
...
+  ${({ borderBottom }) =>
+    borderBottom && 'border-bottom: 1px solid transparent;'}
```
Comment (andrew-nowak) [inline packages/components/psammead-brand/src/index.jsx]: The `0.0625rem solid transparent;` could be factored out into a variable

File: packages/components/psammead-brand/src/index.jsx [hunk 5]
```diff
@@ -145,32 +148,21 @@ StyledBrand.defaultProps = {
...
-  const styledBrandProps = {
-    product,
-    serviceLocalisedName,
-    svgHeight,
-    minWidth,
-    maxWidth,
-    svg,
-  };
+const Brand = props => {
+  const { svgHeight, maxWidth, minWidth, url, borderTop, borderBottom } = props;
```
Comment (andrew-nowak) [inline packages/components/psammead-brand/src/index.jsx:161]: This is nice! I think this looks a lot better than unpacking and repacking `props` :+1:

File: packages/components/psammead-brand/src/index.stories.jsx [hunk 6]
```diff
@@ -55,6 +55,7 @@ storiesOf('Components|Brand', module)
           minWidth={minWidthInput}
           maxWidth={maxWidthInput}
           svg={svgs[svgChoice]}
+          borderBottom
```
Comment (andrew-nowak) [inline packages/components/psammead-brand/src/index.stories.jsx]: Can this be controlled using knobs? I'd like to be able to set `borderTop` and `borderBottom`

**Review by andrew-nowak — CHANGES_REQUESTED** [review]

**Review by andrew-nowak — APPROVED** [review]
cool :+1:

**Review by DenisHdz — APPROVED** [review]
Nice 👍🏻

**Review by dr3 — APPROVED** [review]

File: packages/components/psammead-brand/README.md [hunk 7]
```diff
@@ -34,11 +34,15 @@ The `url` value is the link that points to the frontpage of the service associat
...
 | url | String | no | N/A | `https://www.bbc.co.uk/news` |
 | serviceLocalisedName | String | no | N/A | `'Yoruba'` |
+| borderTop | bool | no | `false` | `true` |
+| borderBottom | bool | no | `false` | `true` |
 
 ## Usage
 
 The typical use-case of this component is at the top of pages in a [`header` element](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/header). When this is done it is recommend that the component is wrapped in a [`banner` role](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/Banner_role). However, a `header` with a `banner` role should only appear once on a page.
 
+When using `Brand` in the header, you should ensure that `borderBottom` prop is set to true. Similarly, when using brand on the footer you should set `borderTop` to true. This ensures that in high contrast mode the top/bottom of the `Brand` component is visible.
```
Comment (greenc05) [inline packages/components/psammead-brand/README.md]: This is also for when a user changes their colour preferences, please could we tweak the text? '... This ensures when in High Contrast Mode on PC and when the user changes colour preferences in FireFox that the top/bottom...

**Comment by DenisHdz** [comment]
This PR has now conflicts.

**Comment by WeimingFan** [comment]
Tom and I have checked over and all looks good, thanks

**Comment by david-boydell** [comment]
Looks good to me, this is ready for merge.

image

**Pull request merged**